### PR TITLE
Update parmed converter to use new residue infrastructure

### PR DESCRIPTION
**Pull request by daico007**
Update parmed conversion method to utilize the new `site.residue_name` and `site.residue_number`.

### CODE REVIEW

**Comment by daico007** [comment]
Need to fix the issue when parmed periodic improper converted into gmso proper dihedral

File: gmso/external/convert_parmed.py [hunk 1]
```diff
@@ -452,43 +456,32 @@ def to_parmed(top, refer_type=True):
...
         ).value
 
         # Add atom to structure
-        structure.add_atom(pmd_atom, resname=residue_name, resnum=residue_idx)
+        if site.residue_name:
+            structure.add_atom(
+                pmd_atom, resname=site.residue_name, resnum=site.residue_number
+            )
+        else:
+            structure.add_atom(pmd_atom, resname="RES", resnum=0)
```
Comment (CalCraven) [inline gmso/external/convert_parmed.py:484]: Do we want to default our resnum to -1 like [parmed does?](https://github.com/ParmEd/ParmEd/blob/5e48857403ce8fd7848abed4c58fbf677781f97b/parmed/topologyobjects.py#L4230)

File: gmso/external/convert_parmed.py [hunk 2]
```diff
@@ -452,43 +456,32 @@ def to_parmed(top, refer_type=True):
...
 
     # Set up unparametrized system
-    # Build subtop_map (site -> top)
-    default_residue = pmd.Residue("RES")
-    for subtop in top.subtops:
-        for site in subtop.sites:
-            subtop_map[site] = subtop
-
     # Build up atom
+    residue_map = dict()
```
Comment (CalCraven) [inline gmso/external/convert_parmed.py:460]: Do we use this residue map anywhere?

File: gmso/external/convert_parmed.py [hunk 3]
```diff
@@ -452,43 +456,32 @@ def to_parmed(top, refer_type=True):
...
+            mass=site.mass.to(u.amu).value,
+            charge=site.charge.to(u.elementary_charge).value,
         )
         pmd_atom.xx, pmd_atom.xy, pmd_atom.xz = site.position.to(
             "angstrom"
         ).value
 
         # Add atom to structure
-        structure.add_atom(pmd_atom, resname=residue_name, resnum=residue_idx)
+        if site.residue_name:
```
Comment (CalCraven) [inline gmso/external/convert_parmed.py:479]: So does this also append the atom to the correct [ResidueList](https://github.com/ParmEd/ParmEd/blob/5e48857403ce8fd7848abed4c58fbf677781f97b/parmed/topologyobjects.py#L4479) in the parmed structure? Or does this just tag the atom with the residue information? Might want to add a unit test to check that.

**Review by CalCraven — CHANGES_REQUESTED** [review]
Just needs unit tests and it's good to go!

File: gmso/external/convert_parmed.py (hunk 3 above)
Comment (daico007) [reply gmso/external/convert_parmed.py:478]: This is handled a few lines below in `structure.rsidues.claim()` (so the residue list is updated)

File: gmso/external/convert_parmed.py (hunk 1 above)
Comment (daico007) [reply gmso/external/convert_parmed.py]: changed as requested.

**Review by justinGilmer — APPROVED** [review]

**Pull request merged**